src/main/java/org/asamk/signal/manager/Manager.java

```diff
@@ -745,6 +725,138 @@ public void setExpirationTimer(String numberOrGroupId, int messageExpirationTime
         account.getThreadStore().updateThread(thread);
     }
 
+    public String uploadStickerPack(String path) throws IOException, StickerPackInvalidException {
```

Comment:
Nice work!

There's a bit too much logic for the sticker upload in this method for my taste.
I'd prefer to a have method in libsignal-service, that takes a data structure (similar to `SignalServiceStickerManifest` class, but with additional InputStream fields for the sticker image data).
So signal-cli would just parse the zip/files into that class and libsignal-service would handle the upload of the manifest and sticker files internally.
What do you think?